src/praisonai/praisonai/bots/botos.py

```
@@ -247,6 +264,127 @@ async def start_bot(name: str, bot_instance: Bot) -> None:
+        Conservative: any due scheduled job keeps the gateway awake so a
+        cron turn is never dropped by suspending mid-run.
+        """
+        try:
+            from praisonaiagents.tools.schedule_tools import _get_store
+            from praisonaiagents.scheduler import ScheduleRunner
+        except ImportError:
+            return False
+        try:
+            runner = ScheduleRunner(_get_store())
+            return bool(runner.get_due_jobs())
+        except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)